---
title: "Sup1_AOU_genetics"
author: "Joy_Fu"
date: "2023-06-15"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list = ls())
# lapply(paste('package:', names(sessionInfo()$otherPkgs), sep = ""),
#        detach, character.only = TRUE, unload = TRUE)
pacman::p_load(tidyverse, bigsnpr, bigreadr, data.table)
raw_data_path = "/Users/Mingzhou/Desktop/Projects/Dementia-prediction/data/"
output_path = "/Users/Mingzhou/Desktop/Projects/Dementia-prediction/output/"
# Source in useful functions
source("/Users/Mingzhou/Desktop/Projects/Dementia-prediction/code/functions.R")
```

# Part 1. Data preparation
## 1. Extract info from AOU database
```{r}
load(paste0(output_path, "AOU_validation/final_eligible_AMR.rda"))
final_eligible_AMR = final_eligible_AOU
dim(final_eligible_AMR) # dim = (3066,8)
sample_id_AMR = final_eligible_AMR %>% 
  select(PatientID) %>% dplyr::rename("sample_id" = "PatientID")
load(paste0(output_path, "AOU_validation/final_eligible_AFR.rda"))
final_eligible_AFR = final_eligible_AOU
dim(final_eligible_AFR) # dim = (4778,8)
sample_id_AFR = final_eligible_AFR %>% 
  select(PatientID) %>% dplyr::rename("sample_id" = "PatientID")

output_sample_id = rbind(sample_id_AMR, sample_id_AFR)
dim(output_sample_id) # dim = (7844,1)
# output to txt
write.table(output_sample_id,
            file = paste0(output_path, "AOU_validation/sample_id.txt"),
            sep = "\t", quote = F, row.names = F, col.names = T)
```

```{r}
options(scipen = 999)
load(file = paste0(raw_data_path, "FUMA_outputs/overlap_fuma.rda"))
dim(overlap_fuma) # dim = (8727, 20)
output_snps = overlap_fuma %>% mutate(stop = as.numeric(start)+1) %>% 
  select(chr, start, stop, ref, alt) %>% 
  # add APOE SNPs
  add_row(chr = "chr19", start = 44908684, stop = 44908685, 
          ref = "T", alt = "C") %>%
  add_row(chr = "chr19", start = 44908822, stop = 44908823, 
          ref = "C", alt = "T") %>% unique()
dim(output_snps) # dim = (7393, 5)
# output to txt
write.table(output_snps, 
            file = paste0(output_path, 'AOU_validation/output_snps.txt'),
            sep = "\t", quote = F, row.names = F, col.names = T)
```

## 2. Data cleaning from AOU outputs
### Sample ancestry PCs
```{r}
# read in txt file
sample_ancestry = read.table(file = paste0(raw_data_path, "AOU/ancestry_pcs.txt"),
                             sep = "\t", header = T, stringsAsFactors = F)
# combine with phenotype info
AOU_AMR = final_eligible_AMR %>% 
  left_join(sample_ancestry, by = c("PatientID" = "research_id")) %>% 
  select(-ancestry_pred)
dim(AOU_AMR) # dim = (3066,24)
AOU_AFR = final_eligible_AFR %>% 
  left_join(sample_ancestry, by = c("PatientID" = "research_id")) %>% 
  select(-ancestry_pred)
dim(AOU_AFR) # dim = (4778,24)
```

### Geno data preparation
```{r}
geno_file = paste0(raw_data_path, 'AOU/model_snps_plink')
snp_readBed(paste0(geno_file, '.bed'))
obj.bigSNP = snp_attach(paste0(geno_file, '.rds'))
# extract the SNP information from the genotype
map = obj.bigSNP$map[-3]
names(map) = c("chr", "rsid", "pos", "a1", "a0")
map = map %>% mutate(chr = as.numeric(substr(chr, 4, nchar(chr))))
# Get fam order
fam.order = as.data.table(obj.bigSNP$fam)
# get genotype info
genotype = obj.bigSNP$genotypes
# genotype.imp = snp_fastImputeSimple(obj.bigSNP$genotypes, method = 'random') 
genotype_df = as.matrix(genotype[])[,] %>% as.data.frame()
colnames(genotype_df) = map$rsid
dim(genotype_df) # dim = (7844,7393)
# check for missing
missing_info = genotype_df %>% 
  summarise_all(function(x) sum(is.na(x))) %>%  
  gather(variable, missing_count) 
# combine together
genotype_df_fam = cbind(fam.order[,2], genotype_df) %>% as.data.frame() 
dim(genotype_df_fam) # dim = (7844,7394)
```

# Part 2. Genetic predictors
## 1. PRSs
```{r}
load(file = paste0(raw_data_path, "modeling/beta_merge_raw.rda"))
```

### A. AD EUR
#### Independent sig SNPs
```{r}
snp_AD_EUR = beta_merge_raw %>% filter(!is.na(beta_EUR_indsig)) %>% 
  select(chr, pos, a0, a1, beta_EUR_indsig) %>% 
  dplyr::rename("beta" = "beta_EUR_indsig")
# Reverse beta if necessary by performing SNP matching
info_snp_AD_EUR = snp_match(snp_AD_EUR, map) # 76 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_AD_EUR$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_AD_EUR$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('AD_PRS_EUR_indsig', 'PatientID')
# Normalization (use 1KG EUR as reference)
AD_EUR_1kg_mean = -0.9308247
AD_EUR_1kg_sd = 1.505618
# Apply to ATLAS
AD_EUR_prs_indsig = pred_prs_final %>% drop_na()
dim(AD_EUR_prs_indsig) # dim = (3515,2)
AD_EUR_prs_indsig$AD_PRS_EUR_indsig = 
  (AD_EUR_prs_indsig$AD_PRS_EUR_indsig - AD_EUR_1kg_mean) / AD_EUR_1kg_sd
```

#### Mapped SNPs
```{r}
snp_AD_EUR = beta_merge_raw %>% filter(!is.na(beta_EUR_map)) %>% 
  select(chr, pos, a0, a1, beta_EUR_map) %>% 
  dplyr::rename("beta" = "beta_EUR_map")
# Reverse beta if necessary by performing SNP matching
info_snp_AD_EUR = snp_match(snp_AD_EUR, map) # 75 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_AD_EUR$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_AD_EUR$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('AD_PRS_EUR_map', 'PatientID')
# Normalization (use 1KG EUR as reference)
AD_EUR_1kg_mean = -0.0667567
AD_EUR_1kg_sd = 1.134523
# Apply to ATLAS
AD_EUR_prs_map = pred_prs_final %>% drop_na()
dim(AD_EUR_prs_map) # dim = (2075,2)
AD_EUR_prs_map$AD_PRS_EUR_map = 
  (AD_EUR_prs_map$AD_PRS_EUR_map - AD_EUR_1kg_mean) / AD_EUR_1kg_sd
```

### B. AD AFR
#### Independent sig SNPs
```{r}
snp_AD_AFR = beta_merge_raw %>% filter(!is.na(beta_AFR_indsig)) %>% 
  select(chr, pos, a0, a1, beta_AFR_indsig) %>% 
  dplyr::rename("beta" = "beta_AFR_indsig")
# Reverse beta if necessary by performing SNP matching
info_snp_AD_AFR = snp_match(snp_AD_AFR, map) # 11 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_AD_AFR$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_AD_AFR$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('AD_PRS_AFR_indsig', 'PatientID')
# Normalization (use 1KG EUR as reference)
AD_EUR_1kg_mean = 0.3909557
AD_EUR_1kg_sd = 1.171444
# Apply to ATLAS
AD_AFR_prs_indsig = pred_prs_final %>% drop_na()
dim(AD_AFR_prs_indsig) # dim = (6521,2)
AD_AFR_prs_indsig$AD_PRS_AFR_indsig = 
  (AD_AFR_prs_indsig$AD_PRS_AFR_indsig - AD_EUR_1kg_mean) / AD_EUR_1kg_sd
```

#### Mapped SNPs
```{r}
snp_AD_AFR = beta_merge_raw %>% filter(!is.na(beta_AFR_map)) %>% 
  select(chr, pos, a0, a1, beta_AFR_map) %>% 
  dplyr::rename("beta" = "beta_AFR_map")
# Reverse beta if necessary by performing SNP matching
info_snp_AD_AFR = snp_match(snp_AD_AFR, map) # 11 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_AD_AFR$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_AD_AFR$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('AD_PRS_AFR_map', 'PatientID')
# Normalization (use 1KG EUR as reference)
AD_EUR_1kg_mean = 0.1592197
AD_EUR_1kg_sd = 0.752526
# Apply to ATLAS
AD_AFR_prs_map = pred_prs_final %>% drop_na()
dim(AD_AFR_prs_map) # dim = (7718,2)
AD_AFR_prs_map$AD_PRS_AFR_map = 
  (AD_AFR_prs_map$AD_PRS_AFR_map - AD_EUR_1kg_mean) / AD_EUR_1kg_sd
```

### C. AD trans
#### Independent sig SNPs
```{r}
snp_AD_trans = beta_merge_raw %>% filter(!is.na(beta_trans_indsig)) %>% 
  select(chr, pos, a0, a1, beta_trans_indsig) %>% 
  dplyr::rename("beta" = "beta_trans_indsig")
# Reverse beta if necessary by performing SNP matching
info_snp_AD_trans = snp_match(snp_AD_trans, map) # 54 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_AD_trans$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_AD_trans$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('AD_PRS_trans_indsig', 'PatientID')
# Normalization (use 1KG EUR as reference)
AD_EUR_1kg_mean = -0.04261034
AD_EUR_1kg_sd = 2.248376
# Apply to ATLAS
AD_trans_prs_indsig = pred_prs_final %>% drop_na()
dim(AD_trans_prs_indsig) # dim = (5314,2)
AD_trans_prs_indsig$AD_PRS_trans_indsig = 
  (AD_trans_prs_indsig$AD_PRS_trans_indsig - AD_EUR_1kg_mean) / AD_EUR_1kg_sd
```

#### Mapped SNPs
```{r}
snp_AD_trans = beta_merge_raw %>% filter(!is.na(beta_trans_map)) %>% 
  select(chr, pos, a0, a1, beta_trans_map) %>% 
  dplyr::rename("beta" = "beta_trans_map")
# Reverse beta if necessary by performing SNP matching
info_snp_AD_trans = snp_match(snp_AD_trans, map) # 54 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_AD_trans$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_AD_trans$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('AD_PRS_trans_map', 'PatientID')
# Normalization (use 1KG EUR as reference)
AD_EUR_1kg_mean = 0.2465559
AD_EUR_1kg_sd = 1.968068
# Apply to ATLAS
AD_trans_prs_map = pred_prs_final %>% drop_na()
dim(AD_trans_prs_map) # dim = (2149,2)
AD_trans_prs_map$AD_PRS_trans_map = 
  (AD_trans_prs_map$AD_PRS_trans_map - AD_EUR_1kg_mean) / AD_EUR_1kg_sd
```

### D. LBD
#### Independent sig SNPs
```{r}
snp_LBD = beta_merge_raw %>% filter(!is.na(beta_LBD_indsig)) %>% 
  select(chr, pos, a0, a1, beta_LBD_indsig) %>% 
  dplyr::rename("beta" = "beta_LBD_indsig")
# Reverse beta if necessary by performing SNP matching
info_snp_LBD = snp_match(snp_LBD, map) # 9 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_LBD$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_LBD$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('LBD_PRS_indsig', 'PatientID')
# Normalization (use 1KG EUR as reference)
LBD_1kg_mean = 0.2926416
LBD_1kg_sd = 0.7824751
# Apply to ATLAS
LBD_prs_indsig = pred_prs_final %>% drop_na()
dim(LBD_prs_indsig) # dim = (7822,2)
LBD_prs_indsig$LBD_PRS_indsig = 
  (LBD_prs_indsig$LBD_PRS_indsig - LBD_1kg_mean) / LBD_1kg_sd
```

#### Mapped SNPs
```{r}
snp_LBD = beta_merge_raw %>% filter(!is.na(beta_LBD_map)) %>% 
  select(chr, pos, a0, a1, beta_LBD_map) %>% 
  dplyr::rename("beta" = "beta_LBD_map")
# Reverse beta if necessary by performing SNP matching
info_snp_LBD = snp_match(snp_LBD, map) # 9 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_LBD$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_LBD$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('LBD_PRS_map', 'PatientID')
# Normalization (use 1KG EUR as reference)
LBD_1kg_mean = -0.05188502
LBD_1kg_sd = 0.6901303
# Apply to ATLAS
LBD_prs_map = pred_prs_final %>% drop_na()
dim(LBD_prs_map) # dim = (7843,2)
LBD_prs_map$LBD_PRS_map = 
  (LBD_prs_map$LBD_PRS_map - LBD_1kg_mean) / LBD_1kg_sd
```

### E. PD
#### Independent sig SNPs
```{r}
snp_PD = beta_merge_raw %>% filter(!is.na(beta_PD_indsig)) %>% 
  select(chr, pos, a0, a1, beta_PD_indsig) %>% 
  dplyr::rename("beta" = "beta_PD_indsig")
# Reverse beta if necessary by performing SNP matching
info_snp_PD = snp_match(snp_PD, map) # 27 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_PD$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_PD$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('PD_PRS_indsig', 'PatientID')
# Normalization (use 1KG EUR as reference)
PD_1kg_mean = -0.4156226
PD_1kg_sd = 0.5244261
# Apply to ATLAS
PD_prs_indsig = pred_prs_final %>% drop_na()
dim(PD_prs_indsig) # dim = (7825,2)
PD_prs_indsig$PD_PRS_indsig = 
  (PD_prs_indsig$PD_PRS_indsig - PD_1kg_mean) / PD_1kg_sd
```

#### Mapped SNPs
```{r}
snp_PD = beta_merge_raw %>% filter(!is.na(beta_PD_map)) %>% 
  select(chr, pos, a0, a1, beta_PD_map) %>% 
  dplyr::rename("beta" = "beta_PD_map")
# Reverse beta if necessary by performing SNP matching
info_snp_PD = snp_match(snp_PD, map) # 27 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_PD$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_PD$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('PD_PRS_map', 'PatientID')
# Normalization (use 1KG EUR as reference)
PD_1kg_mean = -0.2469264
PD_1kg_sd = 0.3098816
# Apply to ATLAS
PD_prs_map = pred_prs_final %>% drop_na()
dim(PD_prs_map) # dim = (5919,2)
PD_prs_map$PD_PRS_map = 
  (PD_prs_map$PD_PRS_map - PD_1kg_mean) / PD_1kg_sd
```

### F. PSP
#### Independent sig SNPs
```{r}
snp_PSP = beta_merge_raw %>% filter(!is.na(beta_PSP_indsig)) %>% 
  select(chr, pos, a0, a1, beta_PSP_indsig) %>% 
  dplyr::rename("beta" = "beta_PSP_indsig")
# Reverse beta if necessary by performing SNP matching
info_snp_PSP = snp_match(snp_PSP, map) # 21 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_PSP$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_PSP$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('PSP_PRS_indsig', 'PatientID')
# Normalization (use 1KG EUR as reference)
PSP_1kg_mean = 0.2032141
PSP_1kg_sd = 0.1415184
# Apply to ATLAS
PSP_prs_indsig = pred_prs_final %>% drop_na()
dim(PSP_prs_indsig) # dim = (6842,2)
PSP_prs_indsig$PSP_PRS_indsig = 
  (PSP_prs_indsig$PSP_PRS_indsig - PSP_1kg_mean) / PSP_1kg_sd
```

#### Mapped SNPs
```{r}
snp_PSP = beta_merge_raw %>% filter(!is.na(beta_PSP_map)) %>% 
  select(chr, pos, a0, a1, beta_PSP_map) %>% 
  dplyr::rename("beta" = "beta_PSP_map")
# Reverse beta if necessary by performing SNP matching
info_snp_PSP = snp_match(snp_PSP, map) # 21 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_PSP$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_PSP$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('PSP_PRS_map', 'PatientID')
# Normalization (use 1KG EUR as reference)
PSP_1kg_mean = 0.07864808
PSP_1kg_sd = 0.1252794
# Apply to ATLAS
PSP_prs_map = pred_prs_final %>% drop_na()
dim(PSP_prs_map) # dim = (6868,2)
PSP_prs_map$PSP_PRS_map = 
  (PSP_prs_map$PSP_PRS_map - PSP_1kg_mean) / PSP_1kg_sd
```

### G. Stroke
#### Independent sig SNPs
```{r}
snp_STROKE = beta_merge_raw %>% filter(!is.na(beta_STROKE_indsig)) %>% 
  select(chr, pos, a0, a1, beta_STROKE_indsig) %>% 
  dplyr::rename("beta" = "beta_STROKE_indsig")
# Reverse beta if necessary by performing SNP matching
info_snp_STROKE = snp_match(snp_STROKE, map) # 21 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_STROKE$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_STROKE$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('STROKE_PRS_indsig', 'PatientID')
# Normalization (use 1KG EUR as reference)
STROKE_1kg_mean = 0.169449
STROKE_1kg_sd = 0.1939814
# Apply to ATLAS
STROKE_prs_indsig = pred_prs_final %>% drop_na()
dim(STROKE_prs_indsig) # dim = (7798,2)
STROKE_prs_indsig$STROKE_PRS_indsig = 
  (STROKE_prs_indsig$STROKE_PRS_indsig - STROKE_1kg_mean) / STROKE_1kg_sd
```

#### Mapped SNPs
```{r}
snp_STROKE = beta_merge_raw %>% filter(!is.na(beta_STROKE_map)) %>% 
  select(chr, pos, a0, a1, beta_STROKE_map) %>% 
  dplyr::rename("beta" = "beta_STROKE_map")
# Reverse beta if necessary by performing SNP matching
info_snp_STROKE = snp_match(snp_STROKE, map) # 20 variants have been matched
# Filter genotype info
genotype.targ.indsig = genotype_df_fam %>% dplyr::select(all_of(info_snp_STROKE$rsid))
genotype.targ.matrix = as.matrix(genotype.targ.indsig[])
# Calculate PRS
pred_prs = genotype.targ.matrix %*% info_snp_STROKE$beta
pred_prs_final = cbind(pred_prs, genotype_df_fam[, 'sample.ID']) %>% as.data.frame()
names(pred_prs_final) = c('STROKE_PRS_map', 'PatientID')
# Normalization (use 1KG EUR as reference)
STROKE_1kg_mean = 0.009353257
STROKE_1kg_sd = 0.1423629
# Apply to ATLAS
STROKE_prs_map = pred_prs_final %>% drop_na()
dim(STROKE_prs_map) # dim = (7807,2)
STROKE_prs_map$STROKE_PRS_map = 
  (STROKE_prs_map$STROKE_PRS_map - STROKE_1kg_mean) / STROKE_1kg_sd
```

### Combine all
```{r}
AOU_AMR_PRS = AOU_AMR %>% 
  left_join(AD_EUR_prs_indsig, by = c("PatientID" = "PatientID")) %>% 
  left_join(AD_EUR_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(AD_AFR_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(AD_AFR_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(AD_trans_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(AD_trans_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(LBD_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(LBD_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(PD_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(PD_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(PSP_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(PSP_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(STROKE_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(STROKE_prs_map, by = c("PatientID" = "PatientID"))
dim(AOU_AMR_PRS) # dim = (3066, 38)
AOU_AFR_PRS = AOU_AFR %>% 
  left_join(AD_EUR_prs_indsig, by = c("PatientID" = "PatientID")) %>% 
  left_join(AD_EUR_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(AD_AFR_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(AD_AFR_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(AD_trans_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(AD_trans_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(LBD_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(LBD_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(PD_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(PD_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(PSP_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(PSP_prs_map, by = c("PatientID" = "PatientID")) %>% 
  left_join(STROKE_prs_indsig, by = c("PatientID" = "PatientID")) %>%
  left_join(STROKE_prs_map, by = c("PatientID" = "PatientID"))
dim(AOU_AFR_PRS) # dim = (4778, 38)
```

## 2. APOE status
```{r}
# chr19:44908684
# define APOE-e4 status
apoe_status = genotype_df_fam %>% 
  select(sample.ID, `chr19:44908684:T:C`, `chr19:44908822:C:T`) %>%
  mutate(e4count = case_when(
    `chr19:44908684:T:C` == 0 & `chr19:44908822:C:T` == 0 ~ 0,
    `chr19:44908684:T:C` == 0 & `chr19:44908822:C:T` == 1 ~ 1,
    `chr19:44908684:T:C` == 0 & `chr19:44908822:C:T` == 2 ~ 2,
    `chr19:44908684:T:C` == 1 & `chr19:44908822:C:T` == 0 ~ 0,
    `chr19:44908684:T:C` == 1 & `chr19:44908822:C:T` == 1 ~ 1,
    `chr19:44908684:T:C` == 1 & `chr19:44908822:C:T` == 2 ~ 1,
    `chr19:44908684:T:C` == 2 & `chr19:44908822:C:T` == 0 ~ 0,
    `chr19:44908684:T:C` == 2 & `chr19:44908822:C:T` == 1 ~ 0,
    `chr19:44908684:T:C` == 2 & `chr19:44908822:C:T` == 2 ~ 0
  )) %>% select(sample.ID, e4count) %>% dplyr::rename("PatientID" = "sample.ID")

AOU_AMR_PRS = AOU_AMR_PRS %>% left_join(apoe_status, by = c("PatientID" = "PatientID"))
AOU_AFR_PRS = AOU_AFR_PRS %>% left_join(apoe_status, by = c("PatientID" = "PatientID"))
```


```{r}
AOU_AMR_PRS = AOU_AMR_PRS %>% 
  dplyr::rename("AD_PRS_Trans_indsig" = "AD_PRS_trans_indsig",
                "AD_PRS_Trans_map" = "AD_PRS_trans_map",
                "Stroke_PRS_indsig" = "STROKE_PRS_indsig",
                "Stroke_PRS_map" = "STROKE_PRS_map")
AOU_AFR_PRS = AOU_AFR_PRS %>%
  dplyr::rename("AD_PRS_Trans_indsig" = "AD_PRS_trans_indsig",
                "AD_PRS_Trans_map" = "AD_PRS_trans_map",
                "Stroke_PRS_indsig" = "STROKE_PRS_indsig",
                "Stroke_PRS_map" = "STROKE_PRS_map")
```


```{r}
# Save
save(AOU_AMR_PRS, file = paste0(raw_data_path, "AOU/AOU_AMR_PRS.rda"))
save(AOU_AFR_PRS, file = paste0(raw_data_path, "AOU/AOU_AFR_PRS.rda"))

aou_geno_freq_full_SNP = genotype_df_fam %>% 
  dplyr::rename("PatientID" = "sample.ID") 
dim(aou_geno_freq_full_SNP) # dim = (7844, 7394)
save(aou_geno_freq_full_SNP, file = paste0(raw_data_path, "AOU/aou_geno_freq_full_SNP.rda"))
```

