---
title: "Sup1_AOU_genetics"
author: "Joy_Fu"
date: "2023-06-15"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list = ls())
lapply(paste('package:', names(sessionInfo()$otherPkgs), sep = ""),
       detach, character.only = TRUE, unload = TRUE)
pacman::p_load(tidyverse, bigsnpr, bigreadr, data.table)
raw_data_path = "/Users/Mingzhou/Desktop/Projects/Dementia-prediction/data/"
output_path = "/Users/Mingzhou/Desktop/Projects/Dementia-prediction/output/"
# Source in useful functions
source("/Users/Mingzhou/Desktop/Projects/Dementia-prediction/code/functions.R")
```

## 1. AD SNPs extraction from UCLA ATLAS
```{r}
# AD trans-indsig PRS SNPs extraction
load(file = paste0(raw_data_path, "prs/mod/AD_trans_beta_map.rda"))
AD_trans_map_snp = AD_trans_beta_map %>% 
  select(rsid, beta_trans_map) %>% 
  separate(rsid, into = c("chr", "pos", "a0", "a1"), 
           sep = ":", remove = FALSE) %>% 
  mutate(pos_plus = as.character(as.numeric(pos) + 1)) %>% 
  mutate(output_var = paste0(chr, ":", pos, "-", pos_plus)) %>% 
  ungroup() %>% unique()
dim(AD_trans_map_snp) # (123,8)
write.table(AD_trans_map_snp,
            file = paste0(output_path, "AOU_validation/AMR/AD_trans_map_snp.txt"),
            sep = "\t", quote = F, row.names = F, col.names = T)
AD_trans_map_snp_short = AD_trans_map_snp %>% 
  select(rsid, chr, pos, a0, a1, beta_trans_map) %>% 
  dplyr::rename("beta" = "beta_trans_map") %>% 
  mutate(pos = as.numeric(pos))
```


## 2. Define AD PRS in AOU
```{r}
geno_file = paste0(raw_data_path, 'AOU/AD_trans_map_snp_plink')
# snp_readBed(paste0(geno_file, '.bed'))
obj.bigSNP = snp_attach(paste0(geno_file, '.rds'))
# extract the SNP information from the genotype
map = obj.bigSNP$map[-3]
names(map) = c("chr", "rsid", "pos", "a1", "a0")
cleaned_snp = snp_match(AD_trans_map_snp_short, map)
# 105 variants have been matched; 0 were flipped and 26 were reversed
column_to_change = cleaned_snp %>% filter(rsid.ss != rsid) %>% pull(rsid) # N = 26
genotype = obj.bigSNP$genotypes
genotype_df = as.matrix(genotype[])[,] %>% as.data.frame()
colnames(genotype_df) = map$rsid
genotype_df_edit = genotype_df %>% 
  mutate(across(.cols = all_of(column_to_change), .fns = ~ abs(. - 2), 
                .names = "{.col}"))
# Add fam info
fam.order = as.data.table(obj.bigSNP$fam)
genotype_df_fam = cbind(fam.order[,2], genotype_df_edit) %>% 
  as.data.frame() %>% drop_na()
colnames(genotype_df_fam)[1] = "IID"
dim(genotype_df_fam) # 53423,123 
# Restrict to selected patients
load(file = paste0(raw_data_path, "AOU/AMR/final_eligible_AOU_amr.rda"))
load(file = paste0(raw_data_path, "AOU/AFR/final_eligible_AOU_afr.rda"))
load(file = paste0(raw_data_path, "AOU/EAS/final_eligible_AOU_eas.rda"))
genotype_df_combined = genotype_df_fam %>% 
  filter(IID %in% final_eligible_AOU_amr$PatientID | 
           IID %in% final_eligible_AOU_afr$PatientID |
           IID %in% final_eligible_AOU_eas$PatientID)
dim(genotype_df_combined) # (4816,124)
missing_info = genotype_df_combined %>% 
  summarise_all(function(x) sum(is.na(x))) %>%  
  gather(variable, missing_count) # No missing!
# Create PRS
genotype.targ.matrix = genotype_df_combined %>% 
  select(all_of(cleaned_snp$rsid)) %>% as.matrix() # dim = (4816,105)
beta = AD_trans_map_snp_short %>% 
  filter(rsid %in% cleaned_snp$rsid.ss) %>% pull(beta)
# Calculate PRS
pred_prs = genotype.targ.matrix %*% beta
pred_prs_final = cbind(genotype_df_AMR[, "IID"], pred_prs) %>% as.data.frame()
names(pred_prs_final) = c('PatientID', 'AD_PRS_trans_map')
# Normalization
load(file = paste0(raw_data_path, "prs/mod/AD_trans_map_EUR_1kg_mean.rda"))
load(file = paste0(raw_data_path, "prs/mod/AD_trans_map_EUR_1kg_sd.rda"))
pred_prs_final$AD_PRS_trans_map_norm = 
  (pred_prs_final$AD_PRS_trans_map - AD_trans_map_EUR_1kg_mean) / AD_trans_map_EUR_1kg_sd
save(pred_prs_final, file = paste0(raw_data_path, "AOU/pred_prs_final.rda"))
```

## 3. APOE status
```{r}
geno_file = paste0(raw_data_path, 'AOU/apoe_snp_plink')
# snp_readBed(paste0(geno_file, '.bed'))
obj.bigSNP = snp_attach(paste0(geno_file, '.rds'))
map = obj.bigSNP$map[-3]
names(map) = c("chr", "rsid", "pos", "a1", "a0")
genotype = obj.bigSNP$genotypes
genotype_df = as.matrix(genotype[])[,] %>% as.data.frame()
colnames(genotype_df) = map$rsid
missing_info = genotype_df %>% 
  summarise_all(function(x) sum(is.na(x))) %>%  
  gather(variable, missing_count) # Only a few missing, we just excluded those people
# Add fam info
fam.order = as.data.table(obj.bigSNP$fam)
genotype_df_fam = cbind(fam.order[,2], genotype_df) %>% 
  as.data.frame() %>% drop_na()
colnames(genotype_df_fam)[1] = "IID"
dim(genotype_df_fam) # 245366,3
# define APOE-e4 status
apoe_status = genotype_df_fam %>% 
  mutate(e4count = case_when(
    `chr19:44908684:T:C` == 0 & `chr19:44908822:C:A` == 0 ~ 0,
    `chr19:44908684:T:C` == 0 & `chr19:44908822:C:A` == 1 ~ 1,
    `chr19:44908684:T:C` == 0 & `chr19:44908822:C:A` == 2 ~ 2,
    `chr19:44908684:T:C` == 1 & `chr19:44908822:C:A` == 0 ~ 0,
    `chr19:44908684:T:C` == 1 & `chr19:44908822:C:A` == 1 ~ 1,
    `chr19:44908684:T:C` == 1 & `chr19:44908822:C:A` == 2 ~ 1,
    `chr19:44908684:T:C` == 2 & `chr19:44908822:C:A` == 0 ~ 0,
    `chr19:44908684:T:C` == 2 & `chr19:44908822:C:A` == 1 ~ 0,
    `chr19:44908684:T:C` == 2 & `chr19:44908822:C:A` == 2 ~ 0
  )) %>% select(IID, e4count) %>% dplyr::rename("PatientID" = "IID")
save(apoe_status, file = paste0(raw_data_path, "AOU/apoe_status.rda"))
```

## 4. LASSO SNPs 
```{r}
load(file = paste0(raw_data_path, "AOU/lasso_map_snp.rda"))
lasso_map_snp = lasso_map_snp %>% mutate(pos = as.numeric(pos))
geno_file = paste0(raw_data_path, 'AOU/lasso_map_snp_plink')
snp_readBed(paste0(geno_file, '.bed'))
obj.bigSNP = snp_attach(paste0(geno_file, '.rds'))
map = obj.bigSNP$map[-3]
names(map) = c("chr", "rsid", "pos", "a1", "a0")
# Check matching
cleaned_snp = snp_match(lasso_map_snp, map)
# 81 variants have been matched; 0 were flipped and 18 were reversed.
column_to_change = cleaned_snp %>% filter(rsid.ss != rsid) %>% pull(rsid)
length(column_to_change) # N = 18
genotype = obj.bigSNP$genotypes
genotype_df = as.matrix(genotype[])[,] %>% as.data.frame()
colnames(genotype_df) = map$rsid
```

```{r}
# Renames columns
genotype_df_edit = genotype_df %>% 
  mutate(across(.cols = all_of(column_to_change), .fns = ~ abs(. - 2), 
                .names = "{.col}")) %>% 
  dplyr::rename("chr1:10730740:A:C" = "chr1:10730740:C:A",
                "chr1:181033435:GT:G" = "chr1:181033435:G:GT",
                "chr1:207629207:C:A" = "chr1:207629207:A:C",
                "chr3:39467477:A:G" = "chr3:39467477:G:A",
                "chr4:15715698:A:G" = "chr4:15715698:G:A",
                "chr4:89715859:G:A" = "chr4:89715859:A:G",
                "chr4:89831054:C:T" = "chr4:89831054:T:C",
                "chr4:89910850:C:T" = "chr4:89910850:T:C",
                "chr4:89953728:C:T" = "chr4:89953728:T:C",
                "chr5:61040177:G:A" = "chr5:61040177:A:G",
                "chr6:45449381:C:T" = "chr6:45449381:T:C",
                "chr11:133897727:T:G" = "chr11:133897727:G:T",
                "chr17:45355435:G:A" = "chr17:45355435:A:G",
                "chr19:44549809:A:C" = "chr19:44549809:C:A",
                "chr19:44696254:C:T" = "chr19:44696254:T:C",
                "chr19:44836881:A:G" = "chr19:44836881:G:A",
                "chr19:44864753:G:A" = "chr19:44864753:A:G",
                "chr19:45027294:G:A" = "chr19:45027294:A:G") %>% 
  select(all_of(cleaned_snp$rsid.ss))
dim(genotype_df_edit) # 245394,81
# combine fam info
fam.order = as.data.table(obj.bigSNP$fam)
lasso_snp = cbind(fam.order$sample.ID, genotype_df_edit) %>% as.data.frame()
names(lasso_snp)[1] = "PatientID"
save(lasso_snp, file = paste0(raw_data_path, "AOU/lasso_snp.rda"))
```



